rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParser.java

```diff
@@ -62,7 +62,7 @@ public class GroovyParser implements Parser {
 
     @Override
     public Stream<SourceFile> parse(@Language("groovy") String... sources) {
-        Pattern packagePattern = Pattern.compile("^package\\s+([^;]+);");
+        Pattern packagePattern = Pattern.compile("\\bpackage\\s+([.\\w]+)");
         Pattern classPattern = Pattern.compile("(class|interface|enum)\\s*(<[^>]*>)?\\s+(\\w+)");
```

Comment:
This ironically makes me wonder if we don't support Groovy trait types in various capacities.
https://groovy-lang.org/objectorientation.html#_traits